src/main/java/model/RacingCars.java

```diff
@@ -0,0 +1,69 @@
+package model;
+
+import exception.ErrorCode;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import service.NumberGenerator;
+
+public class RacingCars {
+
+    private static final int MAX_CAR_COUNT = 100;
+    private final List<Car> cars;
+
+    public RacingCars(final List<Car> cars) {
+        validate(cars);
+        this.cars = cars;
+    }
+
+    public void move(final NumberGenerator generator, final int limit) {
+        for (Car car : cars) {
+            car.move(generator.generateNumber(limit));
+        }
+    }
+
+    public Map<String, Integer> getCurrentPositions() {
+        Map<String, Integer> carResult = new HashMap<>();
+        cars.forEach(car -> carResult.put(car.getName(), car.getCurrentPosition()));
+        return carResult;
+    }
+
+    public List<String> findWinner() {
+        Map<String, Integer> positions = getCurrentPositions();
+        Integer bestScore = Collections.max(positions.values());
+        return positions.entrySet()
+            .stream()
+            .filter(status -> status.getValue().equals(bestScore))
+            .map(Entry::getKey)
+            .collect(Collectors.toList());
+    }
+
+    private void validate(final List<Car> cars) {
+        validateCarDuplicate(cars);
+        validateCarCount(cars.size());
+    }
+
+    private void validateCarDuplicate(final List<Car> cars) {
+        if (isCarDuplicate(cars.stream()
+            .map(Car::getName)
+            .collect(Collectors.toList()))) {
```

Comment:
```suggestion
        List<String> carNames = cars.stream()
            .map(Car::getName)
            .collect(Collectors.toList());
        if (isCarDuplicate(carNames)) {
```
코드를 간결하게 짜는 것도 중요하지만 가독성을 위해 변수로 빼주는 것도 좋습니다.

Reply:
여담으로 변수 또는 메서드 등의 분리를 쉽게 하는 단축키가 있답니다!
[이 영상](https://youtu.be/QYO5_riePOQ)을 참고해두면 인텔리제이를 더욱 편하고 빠르게 사용할 수 있어요. 😉

Reply:
안녕하세요 춘식! 
이렇게 좋은 참고 자료를 주시다니.... 😂  정말 감사합니다!!!!

인텔리제이 장인이 되어서 더 가독성 좋은 코드를 작성할 수 있도록 노력하겠습니다 ㅎ.ㅎ  😉